File: src/main/java/nl/rabobank/kotlinmovement/recipes/model/IngredientRequest.java
package nl.rabobank.kotlinmovement.recipes.model;

import lombok.Data;

import javax.validation.constraints.NotBlank;

@Data
public class IngredientRequest {
    @NotBlank
    private final String name;
    @NotBlank
    private final IngredientType type;
    @NotBlank
    private final int weight;
}

File: java-to-kotlin-complete/src/main/kotlin/nl/rabobank/kotlinmovement/recipes/model/IngredientRequest.kt
package nl.rabobank.kotlinmovement.recipes.model

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class IngredientRequest(
    @field:NotBlank(message = "ingredient.name")
    val name: String?,
    @field:NotNull(message = "ingredient.type")
    val type: IngredientType?,
    @field:NotNull(message = "ingredient.weight")
    val weight: Int?
)
